src/server/app/api/controllers/authController.js

```diff
@@ -1,9 +1,25 @@
 import Boom from 'boom';
 import Bounce from 'bounce';
 import { logErr } from '../../shared/utils';
-import { validateUser } from '../../shared/plugins/auth';
+import { generateToken, validateUser, validateToken } from '../../shared/plugins/auth';
 
 const authController = {
+  async createAuthToken(req, h) {
+    let auth, token;
+    try {
+      if (!req.payload.username) throw new Error("Login requires username.");
+      if (!req.payload.password) throw new Error("Login requires password.");
```

Comment:
Should these errors read "Token generation requires username/password"?

Reply:
The error might be thrown up to the mobile app, so I went for something they would be less confused about (I didn't think most users would know about token generation)